pydantic/env_settings.py

```diff
@@ -66,7 +66,7 @@ def _build_secrets_files(self, _secrets_dir: Union[Path, str, None] = None) -> D
             raise SettingsError(f'secrets_dir must reference a directory, not a {path_type(secrets_path)}')
 
         for field in self.__fields__.values():
-            for env_name in field.field_info.extra['env_names']:
+            for env_name in {field.name}:
```

Comment:
split into 2 commits just to ease review

Reply:
We could for extra caution do `{field.name} | field.field_info.extra['env_names']` to be really sure nothing gets broken but imo it's really a bug that isn't expected